src/hooks/computed.ts

```diff
@@ -0,0 +1,95 @@
+import { tokenQueries } from "@/contexts";
+import { findAllowance, findBalance, findToken } from "@/helpers";
+import { useOracleDataContext } from "@/hooks";
+import type { OracleQueryUI } from "@/types";
+import type { Token } from "@shared/types";
+import type { BigNumber } from "ethers";
+import { useAccount } from "wagmi";
+
+type Computed = {
+  currencyBalance: BigNumber | null;
+  fetchCurrencyBalance?: () => void;
+  currencyAllowance: BigNumber | null;
+  fetchCurrencyAllowance?: () => void;
+  token: Token | null;
+  fetchCurrencyTokenInfo?: () => void;
+  sendCurrencyApprove?: () => void;
+};
+/**
+ * Takes a query and returns computed values based on the query.
+ */
+export function useComputed(query: OracleQueryUI | undefined) {
```

Comment:
i think eventually this should be superceded by adding these queries/mutations to each request